src/Composer/Util/Filesystem.php

```
@@ -251,10 +251,15 @@ public function ensureDirectoryExists(string $directory)
                     $directory.' exists and is not a directory.'
                 );
             }
-            if (!@mkdir($directory, 0777, true)) {
-                throw new \RuntimeException(
-                    $directory.' does not exist and could not be created.'
-                );
+
+            $error = null;
+            if (is_link($directory) && !@$this->unlinkImplementation($directory)) {
+                $error = 'Could not delete symbolic link '.$directory.': '.(error_get_last()['message'] ?? '');
+            }
+
+            $error = @mkdir($directory, 0777, true) ? null : ($error ?? $directory.' does not exist and could not be created: '.(error_get_last()['message'] ?? ''));
```

please use a `if/else` rather than a ternary here. Currently, your `@` suppresses errors for the whole ternary, which is not what we want here as it could hide issues in the composer code.